cloudmock/aws/mockec2/launch_templates.go

```diff
@@ -132,100 +249,16 @@ func (m *MockEC2) CreateLaunchTemplate(request *ec2.CreateLaunchTemplateInput) (
 	if m.LaunchTemplates[id] != nil {
 		return nil, fmt.Errorf("duplicate LaunchTemplateId %s", id)
 	}
-	resp := &ec2.ResponseLaunchTemplateData{
-		DisableApiTermination: request.LaunchTemplateData.DisableApiTermination,
-		EbsOptimized:          request.LaunchTemplateData.EbsOptimized,
-		ImageId:               request.LaunchTemplateData.ImageId,
-		InstanceType:          request.LaunchTemplateData.InstanceType,
-		KeyName:               request.LaunchTemplateData.KeyName,
-		SecurityGroupIds:      request.LaunchTemplateData.SecurityGroupIds,
-		SecurityGroups:        request.LaunchTemplateData.SecurityGroups,
-		UserData:              request.LaunchTemplateData.UserData,
-	}
+
+	resp := createResponseLaunchTemplate(request.LaunchTemplateData)
+
+	m.addTags(id, tagSpecificationsToTags(request.TagSpecifications, ec2.ResourceTypeLaunchTemplate)...)
+
 	m.LaunchTemplates[id] = &launchTemplateInfo{
 		data: resp,
 		name: request.LaunchTemplateName,
 	}
 
```

Comment:
This is the only required change in this file, to send back the request info as a response.
```suggestion
	if request.LaunchTemplateData.MetadataOptions != nil {
		resp.MetadataOptions = &ec2.LaunchTemplateInstanceMetadataOptions{
			HttpTokens:              request.LaunchTemplateData.MetadataOptions.HttpTokens,
			HttpPutResponseHopLimit: request.LaunchTemplateData.MetadataOptions.HttpPutResponseHopLimit,
		}
	}
```

Reply:
@hakman yes your right. My bad. I had gotten a lot of segmentation faults on CreateLaunchTemplateVersion not being present. I had thought that we would require that as we are modifying an existing launch template. 

I have removed the CreateLaunchTemplateVersion function and it works with the metadata options change. Thank you!